File: python/SlidingWindow/BuyAndSellStock.py
"""
121. Best Time to Buy and Sell Stock
Difficulty: Easy
Topics: Array, Dynamic Programming
"""
def maxProfit(prices: list[int]) -> int:
    buy = prices[0]
    profit = 0
    for i in range(1, len(prices)):
        if prices[i] < buy:
            buy = prices[i]
        elif prices[i] - buy > profit:
            profit = prices[i] - buy
    return profit

File: java/SlidingWindow/BuyAndSellStock.java
package java.SlidingWindow;

public class BuyAndSellStock {
    /*
     * 121. Best Time to Buy and Sell Stock
     * Difficulty: Easy
     * Topics: Array, Dynamic Programming
     */
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;
        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice;
            }
        }
        return maxProfit;
    }
}
